types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+Ember.isArray(''); // $ExpectType boolean
+Ember.isArray([]); // $ExpectType boolean
+// isBlank
+Ember.isBlank(''); // $ExpectType boolean
+Ember.isBlank([]); // $ExpectType boolean
+// isEmpty
+Ember.isEmpty(''); // $ExpectType boolean
+Ember.isEmpty([]); // $ExpectType boolean
+// isEqual
+Ember.isEqual('', 'foo'); // $ExpectType boolean
+Ember.isEqual([], ''); // $ExpectType boolean
+// isNone
+Ember.isNone(''); // $ExpectType boolean
+Ember.isNone([]); // $ExpectType boolean
+// isPresent
+Ember.isPresent(''); // $ExpectType boolean
+Ember.isPresent([]); // $ExpectType boolean
+// merge
+Ember.merge({ a: 12 }, { b: 34 }).a; // $ExpectType number
+Ember.merge({}, {});
+// observer
+const o2 = EmberObject.extend({
+    name: 'foo',
+    nameWatcher: Ember.observer('name', () => {}),
+    nameWatcher2: Ember.observer('name', 'fullName', () => {})
+});
+// on
+const o3 = EmberObject.extend({
+    name: 'foo',
+    nameWatcher: Ember.on('init', () => {}),
+    nameWatcher2: Ember.on('destroy', () => {})
+});
+// removeListener
+Ember.addListener(o2, 'create', () => {});
+Ember.addListener({}, 'create', () => {}); // $ExpectError
+// removeObserver
+Ember.removeObserver(o2, 'create', () => {});
+Ember.removeObserver({}, 'create', () => {}); // $ExpectError
+// runInDebug
+Ember.runInDebug(() => {});
+// sendEvent
+Ember.sendEvent(o2, 'clicked', [1, 2]);
+// set
+Ember.set(o2.create(), 'name', 'bar'); // $ExpectType string
+Ember.set(o2.create(), 'nam', 'bar'); // $ExpectError
+// setOwner
+Ember.setOwner(o2.create(), {});
+// setProperties
+Ember.setProperties(o2.create(), { name: 'bar' }).name; // $ExpectType string
+// tryInvoke
+Ember.tryInvoke(o2, 'init');
+Ember.tryInvoke(o2, 'init', [441]);
+// trySet
+Ember.trySet(o2, 'nam', ''); // $ExpectType any
+// typeOf
+Ember.typeOf(''); // $ExpectType "string"
+Ember.typeOf(Ember.A()); // $ExpectType "array"
+// warn
+Ember.warn('be caseful!');
+// VERSION
+Ember.VERSION; // $ExpectType string
+
+// onerror
+
+Ember.onerror = (err: Error) => console.error(err);
+Ember.onerror = (num: number, err: Error) => console.error(err); // $ExpectError
+
+// Classes
+// TODO ContainerProxyMixin
+// Ember
+// Ember.Application
+new Ember.Application(); // $ExpectType Application
+Ember.Application.create(); // $ExpectType Application
+// Ember.ApplicationInstance
+new Ember.ApplicationInstance(); // $ExpectType ApplicationInstance
+Ember.ApplicationInstance.create(); // $ExpectType ApplicationInstance
+// TODO: Ember.ApplicationInstance.BootOptions
+// Ember.Array
+const a1: Ember.Array<string> = [];
+// Ember.ArrayProxy
+new Ember.ArrayProxy<number>([3, 3, 2]); // $ExpectType ArrayProxy<number>
+// Ember.Checkbox
+const cb = new Ember.Checkbox(); // $ExpectType Checkbox
+cb.tagName; // $ExpectType string
+// Ember.Component
+const C1 = Ember.Component.extend({ classNames: ['foo'] });
+class C2 extends Ember.Component {
+    classNames = ['foo'];
+}
+const c1 = new C1();
+const c2 = new C2();
+C1.create();
```

Comment:
missing $ExpectType on all these

Reply:
Another really messy one. If we let this stuff leak into our tests, they'll become very difficult to maintain

![screen shot 2018-09-21 at 10 55 25 am](https://user-images.githubusercontent.com/558005/45897780-ed5a7680-bd8c-11e8-9eca-7f9f6b9eaeb8.png)